app/javascript/components/firmware-registry-add-edit.jsx

```diff
@@ -0,0 +1,135 @@
+    });
+  };
+
+  return (
+    <Grid fluid>
+      <Row>
+        <MiqFormRenderer
+          initialValues={initialValues}
+          schema={
+              {
+                fields: [
+                  {
+                    component: componentTypes.SELECT_COMPONENT,
+                    name: 'type',
+                    options: [{ value: 'FirmwareRegistry::RestApiDepot', label: __('Rest API Depot') }],
+                    label: __('Registry Type'),
+                    placeholder: __('Nothing selected'),
+                    isSearchable: true,
+                    validate: [{
+                      type: validatorTypes.REQUIRED,
+                      message: __('Required'),
+                    }],
+                    validateOnMount: true,
+                    isDisabled: true,
+                  },
+                  {
+                    component: componentTypes.TEXT_FIELD,
+                    name: 'name',
+                    label: __('Name'),
+                    maxLength: 128,
+                    validate: [{
+                      type: validatorTypes.REQUIRED,
+                      message: __('Required'),
+                    }],
+                  },
+                  {
+                    component: componentTypes.TEXT_FIELD,
+                    name: 'url',
+                    label: __('URL'),
+                    placeholder: 'https://example.com:1234/images',
+                    maxLength: 2048,
+                    validate: [{
+                      type: validatorTypes.REQUIRED,
+                      message: __('Required'),
+                    }],
+                  },
+                  {
+                    component: componentTypes.TEXT_FIELD,
+                    name: 'userid',
+                    label: __('Username'),
+                    maxLength: 128,
+                    validate: [{
+                      type: validatorTypes.REQUIRED,
+                      message: __('Required'),
+                    }],
+                  },
+                  {
+                    component: componentTypes.TEXT_FIELD,
+                    name: 'password',
+                    label: __('Password'),
+                    maxLength: 128,
+                    validate: [{
+                      type: validatorTypes.REQUIRED,
+                      message: __('Required'),
+                    }],
+                    type: 'password',
+                  },
+                ],
+              }
+          }
+          onSubmit={() => {}}
+          onCancel={() => {}}
+          buttonsLabels={{
+            submitLabel: __('Create'),
+          }}
+          className=""
+          showFormControls={false}
+          onStateUpdate={handleFormStateUpdate}
+        />
+      </Row>
+    </Grid>
+  );
+};
+
+FirmwareRegistryAddEdit.propTypes = {
+  dispatch: PropTypes.func.isRequired,
+};
+
+export default connect()(FirmwareRegistryAddEdit);
```

Comment:
You are adding unnecessary overhead bu using connect if you are not accessing the state. If you want to use `dispatch` there is a public interface for it at `ManageIQ.redux.store.dispatch`.

Reply:
Sweet, thank you!